src/geojson2h3.js

```diff
@@ -46,7 +46,26 @@ function flatten(arrays) {
             out = arrays[i];
         }
     }
-    return out;
+    return Array.from(new Set(out));
+}
+
+/**
+ * Utility to compute the centroid of a polygon, based on @turf/centroid
+ * @private
+ * @param {Number[][][]} polygon     Polygon, as an array of loops
+ * @return {Number[]} lngLat         Lng/lat centroid
+ */
+function centroid(polygon) {
+    let lngSum = 0;
+    let latSum = 0;
+    let count = 0;
+    const loop = polygon[0];
+    for (let i = 0; i < loop.length; i++) {
+        lngSum += loop[i][0];
+        latSum += loop[i][1];
+        count++;
+    }
+    return [lngSum / count, latSum / count];
```

Comment:
Technically this centroid is weighted based on the *number* of points, so if you have a half-circle attached to a rectangle, it'll be a lot closer to the circle rather than the center of the area defined. This is *probably* not a problem when there's no hexagon that fits inside of the polygon, the hexagon is likely to be larger than the polygon in question and contain the area-based centroid, too, but there may be some situations that is not true.

Reply:
This is a reasonable point, but more complex centroid calculation seemed out of scope here. This follows the lead of https://www.npmjs.com/package/@turf/centroid, which seems reasonable:

> Takes one or more features and calculates the centroid using the mean of all vertices. This lessens the effect of small islands and artifacts when calculating the centroid of a set of polygons.